src/cli/parser.cr

```diff
@@ -36,6 +38,42 @@ module Tanda::CLI
           end
         end
 
+        parser.on("clockin", "Clock in/out") do
+          clock_type : String? = nil
+
+          parser.on("start", "Clock in") do
+            puts "start"
+            clock_type = "start"
+          end
+
+          parser.on("finish", "Clock out") do
+            puts "finish"
+            clock_type = "finish"
+          end
+
+          parser.on("break", "Clock a break") do
+            puts "break"
+            parser.on("start", "Start break") do
+              puts "break start"
+              clock_type = "break_start"
+            end
+
+            parser.on("finish", "Finish break") do
+              puts "break finish"
+              clock_type = "break_finish"
+            end
+          end
+
+          puts clock_type
+          if type = clock_type
+            CLI::Commands::ClockIn.new(client, type).execute
+            exit
+          else
+            Utils::Display.error("You must pass a command to clockin")
+            exit
+          end
```

Comment:
Memo 2 me - The problem is this is executed before any of the above blocks so we exit before the variable ever gets set

Reply:
If you comment this out the puts statements work fine but with it are never hit